### PR TITLE
[WIP] More sensitive prefix autocomplete handling

### PR DESCRIPTION
**Pull request by miccolis**
Get carmen-cache to respect tokenized words & not attempt to complete them for you. See https://github.com/mapbox/carmen/pull/795

Current status; mostly functional. Outstanding tasks;

- [x] Could use a preliminary review
- [x] Depends on rockdb update https://github.com/mapbox/carmen-cache/pull/136
- [x] As currently written this would be a breaking API change. May be worth making it backwards compatible. Input is welcome here...
- [x] Need to at slightly stricter validation of prefix match setup (must be 0,1,2)
- [ ] Add CHANGELOG entry... oh gawd, there is no change log...
- [ ] Regenerate Docs

### CODE REVIEW

**Comment by apendleton** [comment]
Oh but also, looks like this branched from carmen-cache before #132 landed? So... might be a little messy getting all that resolved

File: src/node_util.hpp [hunk 1]
```diff
@@ -114,8 +114,9 @@ inline NAN_METHOD(_genericgetmatching) {
     if (!info[0]->IsString()) {
         return Nan::ThrowTypeError("first arg must be a String");
     }
-    if (!info[1]->IsBoolean()) {
-        return Nan::ThrowTypeError("second arg must be a Bool");
+    if (!info[1]->IsNumber()) {
+        // TODO check value is 0-2
```
[the remark this answers is not in the document]
Comment (miccolis) [reply src/node_util.hpp]: Yea, my question here was whether we'd want to continue to accept a boolean in the node interface to preserve backwards compatibility. I'm leaning towards no and demanding integer 0,1,2

**Comment by miccolis** [comment]
@apendleton thanks for the feedback. My take aways:

- [ ] Add some constants to the JS side to make the integer api business less funny. Use them in tests.
- [ ] Validate that input is actually 0,1,2
- [ ] Add a commend about prefix cache skipping
- [ ] Remove the "Unsure comment"
- [ ] Do the v. fun rebase.

**Comment by miccolis** [comment]
Did the rebase, was kinda ugly. You can see the updated diff at https://github.com/mapbox/carmen-cache/compare/rock-on-n-on...word-boundary

As this depends on the rocksdb update I'll open a new PR from the rebased branch (and close this one) once that lands.

**Comment by miccolis** [comment]
Prefering new PR https://github.com/mapbox/carmen-cache/pull/138

**Pull request closed without merging**